### DISCUSSION TITLE
Not getting Fire Warning alerts from API

### DISCUSSION THREAD

**Discussion by lior-lemonade** [weather-gov/api#524] [Q&A] [answered]
I'm comparing data fetched from the API in the last 6 month and data from archive (https://www.fema.gov/api/open/v1/IpawsArchivedAlerts) and see that alerts with event=Fire Warning aren't returned from the API.

This is the query I'm using: 
https://api.weather.gov/alerts?active=1&severity=Severe,Extreme&certainty=Possible,Likely,Observed&event=Fire%20Warning

Sample event that exists in the archive but wasn't accepted from the API:
https://www.fema.gov/api/open/v1/IpawsArchivedAlerts?$filter=id%20eq%20%276127d0ec434e34d5eec6e042%27

**Comment by formula4** [comment]
I don't have an answer myself (I am a developer) but I understand your confusion. I had been doing extensive testing trying to display all alerts from the API last year to match the map at https://www.spc.noaa.gov/products/wwa/ as much as possible. I had noticed the Hazardous Weather Outlook and Short Term forecast were not returning data. They are listed at https://api.weather.gov/alerts/types. I had sent an email to NCO and was assigned ticket INC0243794. Within a few days, I got an email back saying the problem was "resolved".

Fast forward to the other day and I noticed they weren't still being returned still on the API. I posted here (a few discussions ago) and was told they aren't supposed to be disseminated and it's normal behavior. So I have no idea what was actually "resolved" at the time from NCO's email, and I also don't understand why they are listed on the alerts/types list if they don't return data.

Without writing a book to explain, there seems to be a disconnect somewhere in the system as to what WWAs are actually served. I believe that if they are listed at https://api.weather.gov/alerts/types, there should be no question that it should serve the data. If not, something is broken. Even if it takes a long time to fix it, I am fine with that as long as I am aware. Right now I am confused as to what on that list actually works or if there are some alerts that are listed that just aren't supposed to serve data.  It seems most of it does, but after reading your issue with the Fire Warning it still makes me wonder what is going on, and it makes me not trust the system. As a developer, that is a problem. The problem with some of these advisories is that some are obscure and I can't always be there to verify them against other sources to ensure they are working. Respectfully speaking, its the job of the NWS to ensure that.

I am not trying to nitpick. But what am I supposed to tell my emergency management customers? "Well, most of the advisories work. I am not sure which ones might not display." - That's not really acceptable. 

We are all human beings here and it's nothing personal against any of the devs behind the API whatsoever. I am sure it's complicated behind the scenes. But please consider verifying the WWA's on the alerts/types list and remove the ones that do not return data. With that knowledge, I can go to other sources to fill in these gaps.

**Reply by jah1007** [reply]
Reading through the ticket history it appears this may have been closed in error. @StephenClouse would have a better understanding of the alerts/types endpoint to know if those can be safely removed or if there are plans to use them for something else.

**Comment by jah1007** [comment, marked as answer]
I checked in on the specific fire warning linked above. The reason the API did not see this message is because FEMA specifically blocked transmission to the NWEM channel that the upstream data source uses to pull data from FEMA as noted by:

            {
              "BLOCKCHANNEL": "NWEM"
            },

This message came in to FEMA, it was not natively produced by the Weather Service. Since it's transmission to the Weather Service was blocked, we never received it. Fire Warning is a valid alert type but will only show up on the API if we receive it from FEMA into the upstream data source or produce it ourselves.
